hazelcast/src/main/java/com/hazelcast/map/impl/MapContainer.java

```diff
@@ -285,6 +286,10 @@ public InterceptorRegistry getInterceptorRegistry() {
     // callback called when the MapContainer is de-registered from MapService and destroyed - basically on map-destroy
     public void onDestroy() {
     }
+
+    public boolean shouldCloneOnEntryProcessing() {
+        return getIndexes().hasIndex() & OBJECT.equals(mapConfig.getInMemoryFormat());
+    }
 }
```

Comment:
I know that this is an imperfect location for this method but it's the only common context for all the calls that exists right now. I could also create an EntryProcessorUtils class, but seemed an overkill for now. Feel free to chime in.